### PR TITLE
Add get Auth domain RPC

### PR DESCRIPTION
**Pull request by protocol86**
Adds get Auth domain RPC and related unit tests.

### CODE REVIEW

File: Example/Auth/Tests/FIRGetProjectConfigResponseTests.m [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                          callback:^(FIRGetProjectConfigResponse *_Nullable response,
+                                     NSError *_Nullable error) {
+    callbackInvoked = YES;
+    RPCResponse = response;
+    RPCError = error;
+  }];
+
+  [_RPCIssuer respondWithJSON:@{
+    @"projectId": kTestProjectID,
+    @"authorizedDomains": @[ kTestDomain1, kTestDomain2]
```
Comment (XiangtianDai) [inline Example/Auth/Tests/FIRGetProjectConfigResponseTests.m]: space before ']'

File: Firebase/Auth/Source/RPCs/FIRAuthRPCRequest.h [hunk 2]
```diff
@@ -30,15 +30,22 @@ NS_ASSUME_NONNULL_BEGIN
...
 - (NSURL *)requestURL;
 
+@optional
+
+/** @fn containPostBody
+    @brief Returns whether the request contains a post body or not. Requests without a post body
+        are get requests.
+    @remarks The default implementation returns YES.
+ */
+- (BOOL)containPostBody;
```
Comment (XiangtianDai) [inline Firebase/Auth/Source/RPCs/FIRAuthRPCRequest.h]: `containsPostBody`, like `NSArray`'s `containsObject:`.

Comment (XiangtianDai) [inline Firebase/Auth/Source/RPCs/FIRAuthRPCRequest.h]: Doesn't FIRSecureTokenRequest also need to implement this method?

Comment (protocol86) [inline Firebase/Auth/Source/RPCs/FIRAuthRPCRequest.h]: Yes, added.  I would make it non-optional if I could avoid the warnings.

Comment (protocol86) [inline Firebase/Auth/Source/RPCs/FIRAuthRPCRequest.h]: Done, makes sense and is grammatically correct.

File: Example/Auth/Tests/FIRGetProjectConfigResponseTests.m (hunk 1 above)
Comment (protocol86) [inline Example/Auth/Tests/FIRGetProjectConfigResponseTests.m]: done, thanks.

**Pull request merged**